### PR TITLE
LARP Expansion Pack: Goggles for First Responders, Sol and Mercenaries 

### PR DESCRIPTION
**Pull request by WickedCybs**
Adds variants of the tactical goggles made by HappyFox to a few places. Also fixes a bug related to them not giving flash protection at first.

First Responders can get a pair. These only have a medhud included. A Zeng variant of the goggles was added.

Sol offships can get a pair. These have no HUD. Just flash protection.

Mercenaries can get a pair. No HUD. Just flash protection.

Security had its goggles selection renamed to make it clear it's for sec only, so you will have to reselect it. Sorry.

image
image

### CODE REVIEW

**Comment by WickedCybs** [comment]
!review

File: code/modules/client/preference_setup/loadout/loadout_eyes.dm [hunk 1]
```diff
@@ -147,3 +147,17 @@
 	goggles["goggles, tactical (Zavodskoi)"] = list(/obj/item/clothing/glasses/safety/goggles/goon/zavod, "Zavodskoi Interstellar")
 	goggles["goggles, tactical (Idris)"] = list(/obj/item/clothing/glasses/safety/goggles/goon/idris, "Idris Incorporated")
 	gear_tweaks += new /datum/gear_tweak/path/faction(goggles)
+
+/datum/gear/eyes/medical_goggles
+	display_name = "goggles selection (medical)"
```
Comment (SleepyGemmy) [inline code/modules/client/preference_setup/loadout/loadout_eyes.dm:152]: please make it clear that these are functional and not just regular goggles.
suggestion: "HUD goggles selection (medical)" or "medical HUD goggles selection".

File: code/modules/client/preference_setup/loadout/loadout_eyes.dm [hunk 2]
```diff
@@ -147,3 +147,17 @@
 	goggles["goggles, tactical (Zavodskoi)"] = list(/obj/item/clothing/glasses/safety/goggles/goon/zavod, "Zavodskoi Interstellar")
 	goggles["goggles, tactical (Idris)"] = list(/obj/item/clothing/glasses/safety/goggles/goon/idris, "Idris Incorporated")
 	gear_tweaks += new /datum/gear_tweak/path/faction(goggles)
+
+/datum/gear/eyes/medical_goggles
```
Comment (SleepyGemmy) [inline code/modules/client/preference_setup/loadout/loadout_eyes.dm:151]: ditto.

File: code/modules/client/preference_setup/loadout/loadout_eyes.dm [hunk 3]
```diff
@@ -147,3 +147,17 @@
...
+	display_name = "goggles selection (medical)"
+	description = "A selection of medical goggles. Note that factional ones can only be taken by members of that faction."
+	path = /obj/item/clothing/glasses/safety/goggles/medical
+
+/datum/gear/eyes/medical_goggles/New()
+	allowed_roles = list("First Responder")
```
Comment (SleepyGemmy) [inline code/modules/client/preference_setup/loadout/loadout_eyes.dm:157]: wrong place, goes in `/datum/gear/eyes/medical_goggles`.

File: code/modules/clothing/glasses/glasses.dm [hunk 4]
```diff
@@ -295,15 +296,41 @@ BLIND     // can't see anything
 	flip_down = "up to protect your eyes."
 	flip_up = "and let it hang around your neck."
 
+//essentially just sunglasses
+/obj/item/clothing/glasses/safety/goggles/tactical
+	name = "tactical goggles"
+	desc = "A stylish pair of tactical goggles that protect the eyes from aerosolized chemicals, debris and bright flashes."
```
Comment (SleepyGemmy) [inline code/modules/clothing/glasses/glasses.dm:302]:
```suggestion
	desc = "A stylish pair of tactical goggles that protect the eyes from aerosolized chemicals, debris, and bright flashes."
```

File: code/modules/clothing/glasses/glasses.dm [hunk 5]
```diff
@@ -295,15 +296,41 @@ BLIND     // can't see anything
...
+
+/obj/item/clothing/glasses/safety/goggles/tactical/handle_additional_changes()
+	flash_protection = up ? FLASH_PROTECTION_NONE : FLASH_PROTECTION_MODERATE
+
+/obj/item/clothing/glasses/safety/goggles/tactical/generic
+	sprite_state = "security_goggles"
+
+//security hud
 /obj/item/clothing/glasses/safety/goggles/goon
 	name = "tactical goggles"
```
Comment (SleepyGemmy) [inline code/modules/clothing/glasses/glasses.dm:326]: consider changing name to differentiate these from the "military" ones.

File: code/modules/clothing/glasses/glasses.dm [hunk 6]
```diff
@@ -295,15 +296,41 @@ BLIND     // can't see anything
...
-	desc = "A stylish pair of tactical goggles that protect the eyes from aerosolized chemicals."
+	desc = "A stylish pair of tactical goggles that protect the eyes from aerosolized chemicals, debris and bright flashes."
```
Comment (SleepyGemmy) [inline code/modules/clothing/glasses/glasses.dm:327]:
```suggestion
	desc = "A stylish pair of tactical goggles that protect the eyes from aerosolized chemicals, debris, and bright flashes."
```

File: code/modules/clothing/glasses/glasses.dm [hunk 7]
```diff
@@ -335,6 +362,39 @@ BLIND     // can't see anything
 	sprite_state = "idris_goggles"
 	brand_name = "Idris"
 
+//medical hud
+/obj/item/clothing/glasses/safety/goggles/medical
+	name = "medical goggles"
```
Comment (SleepyGemmy) [inline code/modules/clothing/glasses/glasses.dm:367]:
```suggestion
	name = "medical HUD goggles"
```

**Review by SleepyGemmy — CHANGES_REQUESTED** [review]

File: code/modules/client/preference_setup/loadout/loadout_eyes.dm (hunk 3 above)
Comment (WickedCybs) [reply code/modules/client/preference_setup/loadout/loadout_eyes.dm:157]: It still works. I will move it up though.

File: code/modules/clothing/glasses/glasses.dm (hunk 4 above)
Comment (WickedCybs) [reply code/modules/clothing/glasses/glasses.dm:302]: I don't think serial commas are common here, so I won't be doing that unless we're required to follow American English.

File: code/modules/clothing/glasses/glasses.dm (hunk 5 above)
Comment (WickedCybs) [reply code/modules/clothing/glasses/glasses.dm:326]: Not desired.

File: code/modules/clothing/glasses/glasses.dm (hunk 6 above)
Comment (WickedCybs) [reply code/modules/clothing/glasses/glasses.dm:327]: As the last. I don't think serial commas are common here, so I won't be doing that unless we're required to follow American English.

File: code/modules/clothing/glasses/glasses.dm (hunk 7 above)
Comment (WickedCybs) [reply code/modules/clothing/glasses/glasses.dm:367]: I'll add the fact that it has an HUD to its description as a few other items do. It won't be put into the name, though.

File: code/modules/client/preference_setup/loadout/loadout_eyes.dm [hunk 8]
```diff
@@ -134,16 +134,30 @@
...
 /datum/gear/eyes/goon_goggles
-	display_name = "tactical goggles selection"
+	display_name = "tactical HUD goggles selection (security)"
 	description = "A selection of tactical eyewear. Note that factional ones can only be taken by members of that faction."
 	path = /obj/item/clothing/glasses/safety/goggles/goon
 
 /datum/gear/eyes/goon_goggles/New()
-	allowed_roles = security_positions
 	..()
+	allowed_roles = security_positions
```
Comment (SleepyGemmy) [inline code/modules/client/preference_setup/loadout/loadout_eyes.dm:143]: consider moving this up under `/datum/gear/eyes/goon_goggles`.

Comment (WickedCybs) [reply code/modules/client/preference_setup/loadout/loadout_eyes.dm:143]: It doesn't work if moved above and causes runtime errors.

**Review by SleepyGemmy — APPROVED** [review]
technically fine.

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**